relayer/middleware/missedVaas/missedVaas.middleware.ts

```diff
@@ -223,17 +223,20 @@ export async function tryFetchAndProcess(
   try {
     const isInProgress = await fetchIsInProgress(redis, key, logger);
     if (isInProgress) {
-      // short circuit is missedVaa middleware has already detected this vaa
+      // short circuit if missedVaa middleware has already detected this vaa
       return false;
     }
+
+    // before re-triggering middleware, mark key as in progress to avoid recursion
+    await markInProgress(redis, key, logger);
```

Comment:
Wouldn't this leave the option for a race condition to happen? I didn't look at `markInProgress` but I'd expect something more along the lines of `tryMarkInProgress` where it has a return value that tells you whether it marked the VAA in progress and must process it or it didn't and it's already being processed. I guess `markInProgress` could throw instead in the case that it didn't manage to mark it itself, but I'd prefer something that looks more obviously atomic.

Reply:
What markInProgress does is adding the sequence to redis to mark that the VAA is beeing fetched, so that if another run of the worker tries to reprocess the same VAA, it aborts (See above `if (isInProgress) return false`).
If it fails to mark it in progress the workflow will fetched again, and discarded after when it gets to the handler, so it wouldn't be such a bad thing.

Yup, you are right. If it we succeed in marking it in progress and then we fail to fetch, this workflow won't be retried untill the fetchInProgress expires. I prefered that... if actually this process could run in parallel, the fetch could be called multiple times even if it's in progress (it's much more likely that fetching the VAA takes a long time, than the redis operation). So potentially while the fetch is handled, you could have another fetches beeing started.

In reality, the worker runs using a `while` instead of an event based approach so, though if this good or not is arguable, the fetch could never be repeated, since the next run will never start before this run finishes.

Also, this is the middleware that we'll deprecate :). If we end up leaving the two versions for backward compatibility I'll revert the change

Reply:
Ah, but what I meant was that we may have two "threads" that attempt to mark it in progress. Would only one of those succeed in that case? That's what I find not so obvious. If you say that there's only one invocation of this function running at any particular point in time, I guess that's fine :thinking: